include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -789,6 +789,11 @@ def MmaEncodingTrait : AttrInterface<"MmaEncodingTrait"> {
                     (ins "int":$opIdx,
                          "int":$kWidth)>,
 
+    InterfaceMethod<"Return the number of threads per warp for dot operands.",
```

Comment:
I don't understand the meaning of this method. The operand layout cannot fully be determined based on a given MMA layout, for instance there are multiple operand layouts that would have the same destination layout.

Reply:
This is used for encodings like `dotOpLayout<mmaLayout>`. What's destination layout in your context?

Reply:
So maybe destination layout refers to `mmaLayout`?

Can you provide us with an example where number of threads cannot be determined just based on mma layout and opIdx?

Reply:
right, I meant that for a dot the mma is the destination and the dot_operand are the sources. I guess for Nvidia mma layouts the number of threads is always the same for a given mma version. 
What confused me is that it seems odd to make it a property of the MMA encoding, it made more sense to have it be a property of dot_operand.
Does this help with future simplifications?

Reply:
OH I get your point now.

I was following the convention of other methods like `SmallVector<unsigned> DotOperandEncodingAttr::getRepOrder() const {`. IMO this way we can know the dot operand properties by either calling `DotOperandEncodingAttr::getRepOrder` or `MmaEncodingTrait::getRepOrder`. `MmaEncodingTrait` can be helpful when you don't want to use `dyn_cast` to cast a layout to the exact mma layout.

Reply:
Otherwise we have to implement something like:

```
if (auto mfma = dyn_cast<AMDMfma>(layout)) {
  ...
} else if (auto mma = dyn_cast<NvidiaMma>(layout)) {
  ...
} else if (auto wmma = dyn_cast<AMDWmma>(layout)) {
  ...
}
```